diff --git a/src/fuse_ops.c b/src/fuse_ops.c
index 99c0b4b..5c038ae 100644
--- a/src/fuse_ops.c
+++ b/src/fuse_ops.c
@@ -273,30 +273,17 @@ static int unionfs_ioctl(const char *path, int cmd, void *arg, struct fuse_file_
 		return -ENOSYS;
 #endif
 
-	switch (cmd) {
-	case UNIONFS_ONOFF_DEBUG: {
-		int on_off = *((int *) data);
-		// unionfs-ctl gives the opposite value, so !!
-		bool setRes = set_debug_onoff(!!on_off);
-		if (!setRes)
-			return -EINVAL;
-		return 0;
-	}
-	case UNIONFS_SET_DEBUG_FILE: {
-		char *debug_path = (char *) data;
+	int i = find_rw_branch_cutlast(path);
+	if (i == -1) RETURN(-errno);
+
+	char p[PATHLEN_MAX];
+	if (BUILD_PATH(p, uopt.branches[i].path, path)) RETURN(-ENAMETOOLONG);
 
-		set_debug_path(debug_path, _IOC_SIZE(cmd));
+	int res = ioctl(fi->fh, cmd, data);
 
-		debug_init();
-		return 0;
-	}
-	default:
-		USYSLOG(LOG_ERR, "Unknown ioctl: %d", cmd);
-		return -EINVAL;
-		break;
-	}
+	if (res == -1) RETURN(-errno);
 
-	return 0;
+	RETURN(res);
 }
 #endif
 
